packages/foam-core/src/model/tags.ts

```diff
@@ -3,12 +3,8 @@ import { URI } from './uri';
 import { IDisposable } from '../index';
 import { Resource } from './note';
 
-export type TagMetadata = { uri: URI };
-
 export class FoamTags implements IDisposable {
-  public readonly tags: Map<string, TagMetadata[]> = new Map();
-
-  constructor(private readonly workspace: FoamWorkspace) {}
+  public readonly tags: Map<string, URI[]> = new Map();
```

Comment:
@pderaaij this is the main change I made in your code (everything else was a just a consequence of it)